mod-pubsub-client/src/main/java/org/folio/util/pubsub/PubSubClientUtils.java

```diff
@@ -29,6 +41,128 @@
+   *
+   * @param eventMessage - message with payload and metadata to send
+   * @param params       - okapi connection params
+   * @return - async result with boolean value. True if message was sanded successfully
+   */
+  public static CompletableFuture<Boolean> sendEventMessage(Event eventMessage, OkapiConnectionParams params) {
+    PubsubClient client = new PubsubClient(params.getOkapiUrl(), params.getTenantId(), params.getToken());
+    CompletableFuture<Boolean> result = new CompletableFuture<>();
+    try {
+      client.postPubsubPublish(eventMessage, ar -> {
+        if (ar.statusCode() == HttpStatus.HTTP_NO_CONTENT.toInt()) {
+          result.complete(true);
+        } else {
+          String message = String.format("Error during publishing Event Message in PubSub. Status code: %s . Status message: %s ", ar.statusCode(), ar.statusMessage());
+          LOGGER.error(message);
+          result.completeExceptionally(new EventSendingException(message));
+        }
+      });
+    } catch (Exception e) {
+      result.completeExceptionally(e);
+      return result;
+    }
+    return result;
+  }
+
+  /**
+   * Common method for registering external module in PubSub.
+   *
+   * @param params - okapi connection params
+   * @return - async result with boolean value. True if module was registered successfully
+   */
+  public static CompletableFuture<Boolean> registerModule(OkapiConnectionParams params) {
+    CompletableFuture<Boolean> result = CompletableFuture.completedFuture(false);
+    try {
+      PubsubClient client = new PubsubClient(params.getOkapiUrl(), params.getTenantId(), params.getToken());
+      LOGGER.info("Reading MessagingDescriptor.json");
+      DescriptorHolder descriptorHolder = readMessagingDescriptor();
+      if (descriptorHolder.getPublisherDescriptor() != null) {
+        LOGGER.info("Registering events for publishers");
+        List<EventDescriptor> eventDescriptors = descriptorHolder.getPublisherDescriptor().getEventDescriptors();
+        result = result.thenCompose(ar -> registerEvents(client, eventDescriptors))
+          .thenCompose(ar -> registerPublishers(client, descriptorHolder.getPublisherDescriptor()));
+      }
+      if (descriptorHolder.getSubscriberDescriptor() != null) {
+        result = result.thenCompose(ar -> registerSubscribers(client, descriptorHolder.getSubscriberDescriptor()));
+      }
+      return result;
+    } catch (Exception e) {
+      result.completeExceptionally(e);
+      return result;
+    }
+  }
+
+  private static CompletableFuture<Void> registerEvents(PubsubClient client, List<EventDescriptor> events) {
+    List<CompletableFuture<Boolean>> list = new ArrayList<>();
+    try {
+      for (EventDescriptor eventDescriptor : events) {
+        client.postPubsubEventTypes(null, eventDescriptor, ar -> {
+          if (ar.statusCode() == HttpStatus.HTTP_CREATED.toInt()) {
+            list.add(CompletableFuture.completedFuture(true));
```

Comment:
Add future to the `list` before `postPubsubEventTypes`, then pass the future to the handler to complete or fail it.